bookkeeper-common/src/main/java/org/apache/bookkeeper/common/util/OrderedExecutor.java

```diff
@@ -434,43 +430,46 @@ protected OrderedExecutor(String baseName, int numThreads, ThreadFactory threadF
                 throw new RuntimeException("Couldn't start thread " + i, e);
             }
 
-            // Register gauges
-            statsLogger.scopeLabel("thread", String.valueOf(idx))
-                    .registerGauge(String.format("%s-queue", name), new Gauge<Number>() {
-                @Override
-                public Number getDefaultValue() {
-                    return 0;
-                }
-
-                @Override
-                public Number getSample() {
-                    return thread.getQueue().size();
-                }
-            });
-            statsLogger.scopeLabel("thread", String.valueOf(idx))
-                    .registerGauge(String.format("%s-completed-tasks", name), new Gauge<Number>() {
-                @Override
-                public Number getDefaultValue() {
-                    return 0;
-                }
-
-                @Override
-                public Number getSample() {
-                    return thread.getCompletedTaskCount();
-                }
-            });
-            statsLogger.scopeLabel("thread", String.valueOf(idx))
-                    .registerGauge(String.format("%s-total-tasks", name), new Gauge<Number>() {
-                @Override
-                public Number getDefaultValue() {
-                    return 0;
-                }
-
-                @Override
-                public Number getSample() {
-                    return thread.getTaskCount();
-                }
-            });
+            if (thread instanceof ThreadPoolExecutor) {
```

Comment:
The thread created by `createSingleThreadExecutor` is not a subclass of `ThreadPoolExecutor`, which means we will lose the following thread metrics.

Reply:
Good point. Also, I think there is an opportunity to add more stats around rejected & failed tasks. It would be very useful to see these in the charts.